### PR TITLE
add support for px and % for position things on the ui

### PR DESCRIPTION
**Pull request by gonpombo8**
```tsx
<UiEntity uiTransform={
  position: { top: '1px', left: '10%', right: 10 } ,
  width: '40%',
  height: 10
} />
```

### CODE REVIEW

File: packages/@dcl/react-ecs/src/components/uiTransform/position.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+type PropKey = `${PropName}${Capitalize<keyof Position>}`
+type PropKeyUnit = `${PropName}${Capitalize<keyof Position>}Unit`
+type PositionParsed = {
+  [key in PropKey]?: number
+} & {
+  [key in PropKeyUnit]?: YGUnit
+}
+
+function isPercent(val: PositionUnit) {
+  return val.endsWith('%')
```
Comment (menduz) [inline packages/@dcl/react-ecs/src/components/uiTransform/position.ts:20]:
```suggestion
  return typeof val == 'string' && val.endsWith('%')
```

File: packages/@dcl/react-ecs/src/components/uiTransform/position.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+}
+function isPoint(val: PositionUnit) {
+  return val.endsWith('px')
```
Comment (menduz) [inline packages/@dcl/react-ecs/src/components/uiTransform/position.ts:23]:
```suggestion
  return typeof val == 'string' && val.endsWith('px')
```

File: packages/@dcl/react-ecs/etc/react-ecs.api.md [hunk 3]
```diff
@@ -252,12 +252,17 @@ export interface PBUiTransform {
...
+    top: PositionUnit;
+    right: PositionUnit;
+    bottom: PositionUnit;
+    left: PositionUnit;
 };
 
+// Warning: (ae-missing-release-tag) "PositionUnit" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
+//
+// @public (undocumented)
+export type PositionUnit = `${number}px` | `${number}%`;
```
Comment (menduz) [inline packages/@dcl/react-ecs/etc/react-ecs.api.md:264]:
```suggestion
export type PositionUnit = `${number}px` | `${number}%` | number;
```

File: packages/@dcl/react-ecs/src/components/uiTransform/types.ts [hunk 4]
```diff
@@ -24,14 +24,15 @@ export {
   YGWrap
 }
 
+export type PositionUnit = `${number}px` | `${number}%`
```
Comment (menduz) [inline packages/@dcl/react-ecs/src/components/uiTransform/types.ts:27]:
```suggestion
export type PositionUnit = `${number}px` | `${number}%` | number
```

File: packages/@dcl/react-ecs/src/components/uiTransform/position.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  function getValue(key: 'px' | '%') {
+    return Number(val.slice(0, val.indexOf(key)))
+  }
+  if (isPercent(val)) {
+    return [getValue('%'), YGUnit.YGU_PERCENT]
+  }
+  if (isPoint(val)) {
+    return [getValue('px'), YGUnit.YGU_POINT]
+  }
+  return [0, YGUnit.YGU_UNDEFINED]
```
Comment (menduz) [inline packages/@dcl/react-ecs/src/components/uiTransform/position.ts:36]:
```suggestion
  if (typeof val === 'number') {
    return [val, YGUnit.YGU_POINT]
  }
  return [0, YGUnit.YGU_UNDEFINED]
```

File: test/ecs/react-ui/transform.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    const getDiv = (entity: Entity) => UiTransform.get(entity)
+
+    const position: Position = {
+      top: '1px',
+      left: '2px',
+      right: '3%',
+      bottom: '4px'
+    }
+
+    const ui = () => <UiEntity uiTransform={{ width: 100, position }} />
```
Comment (menduz) [inline test/ecs/react-ui/transform.spec.tsx:33]: Add test with non sanitized values (objects, booleans, strings, objects, null)

**Review by leanmendoza — APPROVED** [review]
amazing
